src/sentry/notifications/notification_action/types.py

```
@@ -197,6 +201,23 @@ def execute_futures(
             for callback, future in futures:
                 safe_execute(callback, job.event, future)
 
+    @staticmethod
+    def send_test_notification(
+        job: WorkflowEventData,
+        futures: Collection[
+            tuple[Callable[[GroupEvent, Sequence[RuleFuture]], None], list[RuleFuture]]
+        ],
+    ) -> None:
+        """
+        This method will execute the futures.
+        Based off of process_rules in post_process.py
+        """
+        with sentry_sdk.start_span(
+            op="workflow_engine.handlers.action.notification.issue_alert.execute_futures"
+        ):
+            for callback, future in futures:
```

don't use `safe_execute` so we can propagate the errors up to the ui

---

i'd argue this is ok because the notification platform will have a better way to handle this and this consolidates this piece of debt within the notification action

---

In the future, I'd like to eliminate `safe_execute` from notifications entirely, but agreed this is a notification platform concern.